tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/SavedModelBundle.java

```diff
@@ -510,21 +508,18 @@ private static SavedModelBundle load(
       TF_Graph graph = TF_NewGraph();
```

Comment:
We probably should call `TF_Graph.newGraph()` here, but I'm not sure...

Reply:
I tried that initially, but `Graph.close()` doesn't use `handle.releaseReference()` like `Session.close()` does, and so it doesn't call the `TF_Graph` deallocator. I think you're the last person to touch both `Graph.delete` and `Session.delete` and I wasn't sure if they were different for a reason.

As I said in the comment above, I think they should be made consistent, but that's going to be a bigger change than just the one which fixes this leak. For example if we switch `Graph` over to use `handle.releaseReference()` and then use `TF_Graph.newGraph` everywhere it'll mean that the `TF_Graph` inside `Graph` is now sensitive to enclosing `PointerScope`s which it currently isn't.

Reply:
Is it that the TF_Graph becomes managed by the TF_Session and we shouldn't touch it anymore after that? It sounds like that from the documentation of the C API. In that case I'd put the call to TF_NewGraph() in TF_Session.loadSessionFromSavedModel() and forget about it.

Reply:
No, I think the graph is separate from the session, or at least that's my reading of the doc here - https://github.com/tensorflow/tensorflow/blob/master/tensorflow/c/c_api.h#L1229. The `TF_NewSession` doc above it suggests the `TF_Graph` is kept alive by the `TF_Session`, but it doesn't say that the session owns the graph, so I think we're still responsible for closing it.